### PR TITLE
Start running Unity tests on CI

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

This adds Windows Mono tests for Unity. This is the only platform we can test for right now because of issues still being sorted out with our self-hosted runners. Other platforms/backends will be added in a follow-up PR.

A few notes:
1. We have 3 failing tests. Fixing them is out of scope for this PR and I'll address them in a follow-up one.
1. PR builds will fail if more than one build tries to build for Unity at the same time. This is being addressed at the runner level and should be fixed at a later point.

Part of https://github.com/realm/realm-dotnet/issues/2406.

### CODE REVIEW

**Comment by nirinchev** [comment]
@LaPeste since @fealebenpae went on a tactical vacation prior to reviewing this PR, you've been volunteered to take his spot 😝

File: .github/templates/test-unity.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    if: ${{ contains(inputs.platform, 'Linux') }}
+    steps:
+      - #@ template.replace(checkoutCode())
+      - name: Download Test Runner
+        uses: #@ actionDownloadArtifact
+        with:
+          name: UnityTestsRunner.linux
+          path: TestRunner
+      - name: Install xvfb
+        run: sudo apt-get install xvfb
```
Comment (LaPeste) [inline .github/templates/test-unity.yml:70]:
```suggestion
        run: sudo apt install -y xvfb
```

**Review by LaPeste — APPROVED** [review]
Looks fine. I just suggested a minuscule change.

**Comment by nirinchev** [comment]
@cla-bot check

**Pull request merged**